Add lxqt desktop group & more

Besanon opened the pull request:
Adding the lxqt desktop with related programs und falkon browser.
Add some initial support for runit init system programs

roddhjav commented:
Thanks for your PR.

It will take a bit of time, as I need to setup a new set of test VM for lxqt. But as it looks similar to xfce it should be easy enough. For some context what distribution are you using?

I will also give a quick overview of your profiles and ask for some initial changes. To sum up what I have seen so far you need to:

- Ensure you respect the [profile guidelines](https://apparmor.pujol.io/development/guidelines/) on all your profile
- Some access need to be restricted. For example:
    - For instance, allowing `/tmp/**` is generally useless and open door for an attacker to leak outside the confinement
    - Never use `@{HOME}/** rw` or `user-write` but `user-write-strict`
    - `@{sys}/devices/@{pci_bus}/**/**/**` they can be restricted a lot.
- You need to use the dbus base abstraction (see https://apparmor.pujol.io/development/dbus/#base) instead of upstream one
- You always get a few various spaces between the rule and the access,(eg: `/usr/share/qt{5,6}/     r,`) it get hard to read, please remove them.
- All profiles need to end with the following rule:  `include if exists <local/-profile-name->`

Regarding `runit` profiles I have some doubts. As the init system is closely related to the rest of the system, adding support for another one is a lot of work and I simply don't have the resources for it.

roddhjav commented on apparmor.d/groups/lxqt/featherpad:
```diff
@@ -0,0 +1,57 @@
+@{exec_path} = @{bin}/featherpad
+profile featherpad @{exec_path} {
+  include <abstractions/base>
+  include <abstractions/common/app>
```
This abstraction must not be used for this kind of profile. See https://apparmor.pujol.io/development/abstractions/#commonapp

roddhjav commented on apparmor.d/groups/lxqt/featherpad:
```diff
@@ -0,0 +1,57 @@
+  include <abstractions/base>
+  include <abstractions/common/app>
+  include <abstractions/gtk>
+  include <abstractions/gvfs-open>
```
The standard way to allow opening resource in this project is to use one of the [child-open](https://github.com/roddhjav/apparmor.d/tree/main/apparmor.d/groups/children) profile. Eg: `  @{open_path}           rPx -> child-open,`

roddhjav commented on apparmor.d/groups/lxqt/featherpad:
```diff
@@ -0,0 +1,57 @@
+  include <abstractions/lxqt>
+  include <abstractions/thumbnails-cache-read>
+  include <abstractions/qt5-shader-cache>
+  include <abstractions/dbus-session-strict>
+  include <abstractions/dbus-accessibility-strict>
```
These abstractions won't work in apparmor.d. You need to use:
```
include <abstractions/bus-accessibility>
include <abstractions/bus-session>
```

Same in any other profiles you use it

roddhjav commented on apparmor.d/groups/lxqt/featherpad:
```diff
@@ -0,0 +1,57 @@
+  include <abstractions/dbus-session-strict>
+  include <abstractions/dbus-accessibility-strict>
+  include <abstractions/gvfs-open>
+  include <abstractions/user-read>
+  include <abstractions/user-write>
```
Use:
```
include <abstractions/user-read-strict>
include <abstractions/user-write-strict>
```

Same in any other profiles you use it

roddhjav commented on apparmor.d/groups/lxqt/featherpad:
```diff
@@ -0,0 +1,57 @@
+
+  @{PROC}/sys/kernel/random/boot_id  		r,
+
+  /var/cache/fontconfig/** rw,  
```
This is handled by either `include <abstractions/fontconfig-cache-read>` or `include <abstractions/fontconfig-cache-write`

Same in any other profiles you use it

roddhjav commented on apparmor.d/groups/lxqt/featherpad:
```diff
@@ -0,0 +1,57 @@
+  owner @{PROC}/@{pid}/mountinfo 		r,
+  owner @{PROC}/@{pid}/mounts 			r,
+
+  @{sys}/devices/@{pci_bus}/{,**}               r,
+  @{sys}/devices/@{pci_bus}/**/**/**            r,
```
This is too large. It need to be restricted.

Same in any other profiles you use it

roddhjav commented on apparmor.d/groups/lxqt/featherpad:
```diff
@@ -0,0 +1,57 @@
+
+  /etc/fstab r,
+
+  /opt/{,**}  r,
```
featherpad is a file editor. I don't think `/opt/{,**}` is required at all.

roddhjav commented on apparmor.d/groups/lxqt/featherpad:
```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2024 Alexandre Pujol <alexandre@pujol.io>
+# Copyright (C) 2024 Besanon  <m231009ts@mailfence.com>
+# SPDX-License-Identifier: GPL-2.0-only
+
+#include <tunables/global>
```
On all profiles, you need to set the abi and use `include` instead of `#include` (this form is deprecated) :
```
abi <abi/3.0>,

include <tunables/global>
```

roddhjav commented on apparmor.d/groups/lxqt/lximage-qt:
```diff
@@ -0,0 +1,64 @@
+
+  /etc/xdg/menus/lxqt-applications.menu r,
+
+  owner @{HOME}/.inputrc 		r,
+  owner @{HOME}/.bashrc 		r,
+  owner @{HOME}/.bash_profile 		r,
+  owner @{HOME}/.bash_logout 		r,
+  owner @{HOME}/.bash_history 		r,
```
None of this should be needed.

roddhjav commented on apparmor.d/groups/lxqt/lximage-qt:
```diff
@@ -0,0 +1,64 @@
+  owner @{user_config_dirs}/#@{int} rwk,
+  owner @{user_config_dirs}/QtProject.conf rw,
+  owner @{user_config_dirs}/QtProject.conf.lock rwk, 
+  owner @{user_config_dirs}/**  rwkl  -> @{user_config_dirs}/#@{int},
```
`@{user_config_dirs}/** rwkl -> @{user_config_dirs}/#@{int},` is **way** too large.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-admin-user:
```diff
@@ -0,0 +1,32 @@
+  
+  @{exec_path}			mr,
+
+   /etc/passwd  			r,
+   /etc/group   			r,
+   /etc/shells 				r, 
+   /etc/nsswitch.conf   		r,
```
Use `include <abstractions/nameservice-strict>` instead of these rules,

roddhjav commented on apparmor.d/groups/lxqt/lxqt-admin-user-helper:
```diff
@@ -0,0 +1,24 @@
+    
+  @{exec_path}		mr,
+  
+  owner /tmp/*  		r,
```
This rule is too wide. It needs to be restricted.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-appearance:
```diff
@@ -0,0 +1,39 @@
+  /usr/share/libdrm/amdgpu.ids 		r,
+
+  owner @{user_config_dirs}/lxqt/** rwkl -> @{user_config_dirs}/lxqt/#@{int},
+  owner @{user_config_dirs}/dconf/user  r,
```
Simply use `include <abstractions/dconf-write>`

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-appearance:
```diff
@@ -0,0 +1,39 @@
+  owner @{user_config_dirs}/pcmanfm-qt/lxqt/settings.conf r,
+
+  owner /tmp/#@{int} 			rw,
+  owner /tmp/** rwkl -> /tmp/#@{int},
```
This rule is too wide. It needs to be restricted.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-input:
```diff
@@ -0,0 +1,67 @@
+
+  owner /tmp/@{int}  		       r,
+
+  owner @{run}/udev/data/ r,
+  owner @{run}/udev/data/** r,
+  owner @{run}/udev/data/**/** r,
```
Remove this. It is too wide. And you already have a restricted version.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-input:
```diff
@@ -0,0 +1,67 @@
+  @{run}/udev/data/c@{dynamic}:@{int} r,  # For dynamic assignment range 234 to 254, 384 to 511
+
+  @{sys}/devices/	               r,
+  @{sys}/devices/{,**}     	       r,
+  @{sys}/class/ 		       r,
+  @{sys}/class/{,**} 		       r,
+  @{sys}/bus/ 			       r,
+  @{sys}/bus/{,**}		       r,
+
```
Too wide.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-locale:
```diff
@@ -0,0 +1,34 @@
+  @{exec_path}			mr,
+
+  owner @{user_config_dirs}/lxqt/*  	r,
+  owner @{user_config_dirs}/lxqt/lxqt-config-locale.conf.lock rwk,
```
Already included by the next rule.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-config-notificationd:
```diff
@@ -0,0 +1,35 @@
+  include <abstractions/dbus-accessibility-strict>
+  include <abstractions/gvfs-open>
+    
+  /usr/share/libdrm/amdgpu.ids  	r,
```
Should come after `@{exec_path} mr,`

roddhjav commented on apparmor.d/groups/lxqt/lxqt-session:
```diff
@@ -0,0 +1,59 @@
+
+  /dev/tty						rw,
+  
+  dbus receive
+  	bus=session
+  	path="/org/freedesktop/Notifications"
+  	interface="org.freedesktop.DBus.Introspectable"
+  	peer=(name=":[0-9]*.[0-9]*"),
+  dbus send
+  	bus=session
+  	path="/org/freedesktop/Notifications"
+  	interface="org.freedesktop.Notifications"
+  	peer=(name="org.freedesktop.DBus"),
+  dbus receive
+  	bus=session
+  	path="/org/freedesktop/Notifications"
+  	interface="org.freedesktop.Notifications"
+  	peer=(name=":[0-9]*.[0-9]*"),
```
Use the directive: `aa:dbus own bus=session name=org.freedesktop.Notifications`

Besanon replied:
hmm, if i delete the dbus receive and send statements and replace it with
aa:dbus own bus=session name=org.freedesktop.Notifications
it states that this is unknown.
Havent worked with aa:... statements yet - or have i misunderstood what to change?

roddhjav replied:
Forgot to add the comment, it should be: `#aa:dbus own bus=session name=org.freedesktop.Notifications`

roddhjav commented on apparmor.d/groups/lxqt/obconf-qt:
```diff
@@ -0,0 +1,50 @@
+
+  /usr/share/obconf/{,*} 			r,
+  /usr/share/ibus/setup/{,**}  			rw,
+  /usr/share/drirc.d/ 				r,
+  /usr/share/drirc.d/00-mesa-defaults.conf	r,
+  /usr/share/drirc.d/00-radv-defaults.conf	r,
```
Remove as well as:
```
  /dev/dri/card@{int}				rw,
  owner /dev/tty@{int} 				rw,
```

and use  `include <abstractions/graphics>`.

roddhjav commented on apparmor.d/abstractions/lxqt:
```diff
@@ -0,0 +1,30 @@
+
+  include <abstractions/fonts>
+  include <abstractions/freedesktop.org>
+  include <abstractions/qt5>
```
Also add `include <abstractions/gtk>`

roddhjav commented on apparmor.d/abstractions/lxqt:
```diff
@@ -0,0 +1,30 @@
+  include <abstractions/fonts>
+  include <abstractions/freedesktop.org>
+  include <abstractions/qt5>
+  include <abstractions/mesa>
```
Remove `include <abstractions/mesa>` it is part of `include <abstractions/graphics>`

roddhjav commented on apparmor.d/abstractions/lxqt:
```diff
@@ -0,0 +1,30 @@
+  /usr/share/qt{5,6}/ 					r,
+  /usr/share/qt{5,6}/{,**} 				r,
+
+  /var/cache/fontconfig/* 				rw,
```
These are handled by one of the fontconfig abstraction. Remove from this one.

roddhjav commented on apparmor.d/groups/browsers/falkon:
```diff
@@ -0,0 +1,204 @@
+  owner @{user_config_dirs}/kdedefaults/* r,
+  owner @{user_config_dirs}/kdeglobals r,
+  owner @{user_config_dirs}/kdeglobals.lock rwk,
+  owner @{user_config_dirs}/** rwkl  -> @{user_config_dirs}/#@{int}, 
```
Remove this.

roddhjav commented on apparmor.d/groups/browsers/falkon:
```diff
@@ -0,0 +1,204 @@
+  owner @{user_config_dirs}/falkon/profiles/** rwkl -> @{user_config_dirs}/falkon/profiles/#@{int},
+  owner @{user_config_dirs}/falkonrc.lock rwk,
+  owner @{user_config_dirs}/chromium/WidevineCdm/**   r,
+  owner @{user_config_dirs}/chromium/WidevineCdm/4.10.2710.0/_platform_specific/linux_x64/*.so  m,
```
You have an hardcoded version number.

roddhjav commented on apparmor.d/groups/browsers/falkon:
```diff
@@ -0,0 +1,204 @@
+  /var/tmp/ r,
+
+  owner @{run}/user/@{uid}/#@{int}  rw, 
+  owner @{run}/user/@{uid}/**  rwkl  -> @{run}/user/@{uid}/#@{int},
```
This is **way** too wide.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-session:
```diff
@@ -0,0 +1,59 @@
+  
+   @{exec_path}				mr,
+
+   /usr/share/libdrm/amdgpu.ids 		r,
```
Remove it, add `<abstractions/graphics>`.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-session:
```diff
@@ -0,0 +1,59 @@
+   owner @{user_cache_dirs}/lxqt-notificationd/**  rwk,
+   owner @{user_cache_dirs}/lxqt-notificationd/#@{int} rw,
+   owner @{user_cache_dirs}/lxqt-notificationd/unattended.list.@{rand6} rwkl -> @{user_cache_dirs}/lxqt-notificationd/#@{int},
+   owner @{user_cache_dirs}/mesa_shader_cache/index rwk,
```
Remove, included in  `<abstractions/graphics>`.

roddhjav commented on apparmor.d/groups/lxqt/qpdfview:
```diff
@@ -0,0 +1,67 @@
+include <tunables/global>
+
+@{exec_path} = @{bin}/qpdfview
+profile qpdfview @{exec_path} {
```
This profile is already included in this project.

roddhjav commented on apparmor.d/groups/runit/runsv:
```diff
@@ -0,0 +1,177 @@
+  capability fowner,
+  capability kill,
+
+  signal (send) peer=runsvdir,
+  signal (send) peer=runit,
+  signal (send) peer=NetworkManager,
+  signal (send) peer=agetty,
+  signal (send) set=(cont, term) peer=login,
+  signal (send) set=(cont, term) peer=dhcpcd,
+  signal (send) set=(cont, term) peer=sddm,
+  signal (send) peer=pause,
+  signal (send) peer=cupsd,
+  signal (send) peer=auditd,
+  signal (send) peer=chronyd,
+  signal (send) peer=dbus-daemon,
+  signal (send) peer=udevd,
+  signal (send) peer=rsyslogd,
+  signal (send) set=(kill, term) peer=startlxqt,
+  signal (send) set=(kill, term) peer=xorg,
+  signal (send) set=(term, cont) peer=@{p_systemd},
+  signal (receive) peer=runit,
+  signal (receive) peer=sddm,
+  signal (send) set=(cont, term) peer=elogind,
+  signal (receive) set=(cont, term) peer=elogind,
+  signal (receive) set=(hup) peer=@{p_systemd},
```
As runsv is starts and manages the services, it is able to send signal everywhere: `signal (send),`

roddhjav commented on apparmor.d/groups/runit/runsv:
```diff
@@ -0,0 +1,177 @@
+
+  @{exec_pathrunsv} 	 mr,
+  
+  @{bin}/sv 		 rPx,
+  @{bin}/vlogger         rPx,
+  @{bin}/udevadm 	 rCx -> udevadm, 
+  @{bin}/tlp             rPx,
+  @{bin}/readlink        rix,
+  @{bin}/ethtool         rix,
+  @{bin}/agetty          rPx,
+  @{bin}/id              rPx,
+  @{bin}/rsyslogd        rPx,
+  @{bin}/iw              rPx,
+  @{bin}/cupsd           rPx,
+  @{bin}/dhcpcd          rPx,
+  @{bin}/udevd           rPx,
+  @{bin}/dbus-daemon     rPx,
+  @{bin}/auditd          rPx,
+  @{bin}/chronyd         rPx,
+  @{bin}/NetworkManager  rPx,
+  @{bin}/mount  	 rPx,
+  @{bin}/sddm            rPx,
+  @{bin}/pause           rix,
+  @{bin}/install         rix,
+  @{bin}/chpst           rPx,
+  @{bin}/mkdir           rix,
+  @{bin}/mktemp          rix,
+  @{bin}/dbus-send       rix, # alt: rix
+  @{bin}/utmpset	 rix, # alt: rix
+  @{lib}exec/elogind/elogind          rPx,
+  @{lib}exec/elogind/elogind.wrapper  rPx,  # alt: rix
+  @{bin}/bash		 rPx, # alt: rix
+  @{bin}/tr		 rPx, # alt:rix
+  @{bin}/rm		 rix,
+  @{bin}/touch		 rix,
+  @{bin}/flock		 rix,
+  @{bin}/cat		 rix,
+  @{bin}/grep		 rPx, # alt:rix
+  @{bin}/mountpoint 	 rix,
+  @{bin}/systemctl       rCx -> systemctl,
```
As the service manager, it can start anything:
```
  @{bin}/**                          PUx,
  @{lib}/**                           PUx,
  /usr/share/*/**                   PUx,
```

Besanon replied:
Shouldn´t that be left for runit-init ?

roddhjav replied:
I don't know runit. Therefore I can't says which process is doing what. But looking at this list, to me it looks like the list of service started by your system.

Besanon commented:
Hello Alex,

I am using Archlinux 6/2024 Image and Void Linux.
I will have a look at all your replies and add/alter the statements, havent followed all rules so far. The confinement isnt this close, i agree - although the processes  except runit-init are indeed already confined. Yes, you have to rebuild a lot . Runit is a very condensed init system and it took me several days to construct the few profiles to work with all the others - including some hundred from you. The statements are differnt if you choose sddm or just use dbus-run-session to start the desktop..
The confinement of a whole init system is perhaps a bit too much for me, but step by step will do a good wotrk over time.
Today i have read about your advice not to confine some basic programs - we should use toolbx instead. Sadly... toolbx depends on systemd(!) Is there any alternative to be used on non-systemd environments?
Would be great to have another software for these other init environments. Well, most of the differences regarding the init systems are in the dbus and init programs handling of the Desktop. The session profiles have to be altered. But thats it. Everything else in yur profiles is by far (>95%) working. A severe difference is udiskd which works enforced in Arch but not in Void. I think i will open up another repository which will offer only all profiles which have to be different from your collection to run ALL profiles in the runit init context (so far about 700...750 profiles are running).

Thank you very much for your attention and comments! It will be very useful even for the profiles which will be used in anther init context! 

Regards 

Am 05.06.2024 um 18:23, Alex ***@***.***> schrieb:

Thanks for your PR.
It will take a bit of time, as I need to setup a new set of test VM for lxqt. But as it looks similar to xfce it should be easy enough. For some context what distribution are you using?
I will also give a quick overview of your profiles and ask for some initial changes. To sum up what I have seen so far you need to:Ensure you respect the profile guidelines on all your profileSome access need to be restricted. For example:For instance, allowing /tmp/** is generally useless and open door for an attacker to leak outside the confinementNever use @{HOME}/** rw or user-write but user-write-strict@{sys}/devices/@{pci_bus}/**/**/** they can be restricted a lot.You need to use the dbus base abstraction (see https://apparmor.pujol.io/development/dbus/#base) instead of upstream oneYou always get a few various spaces between the rule and the access,(eg: /usr/share/qt{5,6}/     r,) it get hard to read, please remove them.All profiles need to end with the following rule:  include if exists <local/-profile-name->
Regarding runit profiles I have some doubts. As the init system is closely related to the rest of the system, adding support for another one is a lot of work and I simply don't have the resources for it.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

-- 
Sent with https://mailfence.com  
Secure and private email

Besanon commented on apparmor.d/groups/LXQT:
can be removed

Besanon commented:
Hello Alex,

there are two points i noticed right from the start in Arch linux (hardened kernel) setup:

dbus                 wants to exec dbus-update-activation-environment
sddm-greeter    wants to receive signal from sddm

Regards

Am 05.06.2024 um 18:23, Alex ***@***.***> schrieb:

Thanks for your PR.
It will take a bit of time, as I need to setup a new set of test VM for lxqt. But as it looks similar to xfce it should be easy enough. For some context what distribution are you using?
I will also give a quick overview of your profiles and ask for some initial changes. To sum up what I have seen so far you need to:Ensure you respect the profile guidelines on all your profileSome access need to be restricted. For example:For instance, allowing /tmp/** is generally useless and open door for an attacker to leak outside the confinementNever use @{HOME}/** rw or user-write but user-write-strict@{sys}/devices/@{pci_bus}/**/**/** they can be restricted a lot.You need to use the dbus base abstraction (see https://apparmor.pujol.io/development/dbus/#base) instead of upstream oneYou always get a few various spaces between the rule and the access,(eg: /usr/share/qt{5,6}/     r,) it get hard to read, please remove them.All profiles need to end with the following rule:  include if exists <local/-profile-name->
Regarding runit profiles I have some doubts. As the init system is closely related to the rest of the system, adding support for another one is a lot of work and I simply don't have the resources for it.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

-- 
Sent with https://mailfence.com  
Secure and private email

Besanon commented:
Good Morning,

the falkon and elogind profile are adapted. 

I have added the formal statements for the other profiles lxqt* and added some more relevant profiles.
As far as i can see, there are issues with dbus-update-activation in startlxqt and lxqt-notificationd. All other profiles run without issues after the apatation of the include-statements.
aa-log.txt is the result when running everything in complain mode. aa-log2.txt is the result when disabling lxqt-notifiicationd. aa-log3.txt the result when disabling startlxqt (successful login possible).
continue testing...

Regards,

Am 06.06.2024 um 12:44, Alex ***@***.***> schrieb:

@roddhjav commented on this pull request.
In apparmor.d/groups/lxqt/lxqt-session:> +  dbus receive+  	bus=session+  	path="/org/freedesktop/Notifications"+  	interface="org.freedesktop.DBus.Introspectable"+  	peer=(name=":[0-9]*.[0-9]*"),+  dbus send+  	bus=session+  	path="/org/freedesktop/Notifications"+  	interface="org.freedesktop.Notifications"+  	peer=(name="org.freedesktop.DBus"),+  dbus receive+  	bus=session+  	path="/org/freedesktop/Notifications"+  	interface="org.freedesktop.Notifications"+  	peer=(name=":[0-9]*.[0-9]*"),
Forgot to add the comment, it should be: #aa:dbus own bus=session name=org.freedesktop.Notifications
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

-- 
Sent with https://mailfence.com  
Secure and private email
[1;32mALLOWED[0m [34mauditd[0m [33mcapable[0m info="optional: no audit" comm=auditd error=-1 capname=fowner capability=3
[1;32mALLOWED[0m [34mxorg[0m [33mcapable[0m info="optional: no audit" comm=Xorg capname=fowner error=-1 capability=3
[1;32mALLOWED[0m [34munix-chkpwd[0m [33mfile_mmap[0m[35m owner[0m [35m[0m info="Failed name lookup - disconnected path" comm=unix_chkpwd requested_mask=[1;31mr[0m denied_mask=[1;31mr[0m error=-13
[1;32mALLOWED[0m [34msddm//systemctl[0m [33mfile_inherit[0m[35m owner[0m [35m@{user_share_dirs}/sddm/xorg-session.log[0m comm=systemctl requested_mask=[1;31mw[0m denied_mask=[1;31mw[0m
[1;32mALLOWED[0m [34mlxqt-notificationd[0m [33mfile_receive[0m [35mrun/systemd/inhibit/2.ref[0m info="Failed name lookup - disconnected path" comm=QDBusConnection requested_mask=[1;31mw[0m denied_mask=[1;31mw[0m error=-13

[1;32mALLOWED[0m [34mauditd[0m [33mcapable[0m info="optional: no audit" comm=auditd error=-1 capability=3 capname=fowner
[1;32mALLOWED[0m [34mxorg[0m [33mcapable[0m info="optional: no audit" comm=Xorg error=-1 capability=3 capname=fowner
[1;32mALLOWED[0m [34munix-chkpwd[0m [33mfile_mmap[0m[35m owner[0m [35m[0m info="Failed name lookup - disconnected path" comm=unix_chkpwd requested_mask=[1;31mr[0m denied_mask=[1;31mr[0m error=-13
[1;32mALLOWED[0m [34msddm//systemctl[0m [33mfile_inherit[0m[35m owner[0m [35m@{user_share_dirs}/sddm/xorg-session.log[0m comm=systemctl requested_mask=[1;31mw[0m denied_mask=[1;31mw[0m
[1;32mALLOWED[0m [34msddm[0m [33mexec[0m [35m@{bin}/dbus-update-activation-environment[0m -> [35msddm//null-@{bin}/dbus-update-activation-environment[0m comm=bash requested_mask=[1;31mx[0m denied_mask=[1;31mx[0m
[1;32mALLOWED[0m [34msddm//null-@{bin}/dbus-update-activation-environment[0m [33mfile_inherit[0m[35m owner[0m [35m@{user_share_dirs}/sddm/xorg-session.log[0m comm=dbus-update-act requested_mask=[1;31mw[0m denied_mask=[1;31mw[0m
[1;32mALLOWED[0m [34msddm//null-@{bin}/dbus-update-activation-environment[0m [33mfile_mmap[0m [35m@{bin}/dbus-update-activation-environment[0m comm=dbus-update-act requested_mask=[1;31mr[0m denied_mask=[1;31mr[0m
[1;32mALLOWED[0m [34mstartlxqt[0m [33mexec[0m [35m@{bin}/lxqt-session[0m -> [35m@{bin}/lxqt-session[0m info="profile transition not found" comm=startlxqt requested_mask=[1;31mx[0m denied_mask=[1;31mx[0m error=-13
[1;32mALLOWED[0m [34msddm-greeter[0m [33msignal[0m comm=sddm-helper requested_mask=[1;31mreceive[0m denied_mask=[1;31mreceive[0m signal=term peer=sddm

[1;32mALLOWED[0m [34mauditd[0m [33mcapable[0m info="optional: no audit" comm=auditd error=-1 capability=3 capname=fowner
[1;32mALLOWED[0m [34mxorg[0m [33mcapable[0m info="optional: no audit" comm=Xorg error=-1 capability=3 capname=fowner
[1;32mALLOWED[0m [34munix-chkpwd[0m [33mfile_mmap[0m[35m owner[0m [35m[0m info="Failed name lookup - disconnected path" comm=unix_chkpwd requested_mask=[1;31mr[0m denied_mask=[1;31mr[0m error=-13
[1;32mALLOWED[0m [34msddm//systemctl[0m [33mfile_inherit[0m[35m owner[0m [35m@{user_share_dirs}/sddm/xorg-session.log[0m comm=systemctl requested_mask=[1;31mw[0m denied_mask=[1;31mw[0m
[1;32mALLOWED[0m [34msddm[0m [33mexec[0m [35m@{bin}/dbus-update-activation-environment[0m -> [35msddm//null-@{bin}/dbus-update-activation-environment[0m comm=bash requested_mask=[1;31mx[0m denied_mask=[1;31mx[0m
[1;32mALLOWED[0m [34msddm//null-@{bin}/dbus-update-activation-environment[0m [33mfile_inherit[0m[35m owner[0m [35m@{user_share_dirs}/sddm/xorg-session.log[0m comm=dbus-update-act requested_mask=[1;31mw[0m denied_mask=[1;31mw[0m
[1;32mALLOWED[0m [34msddm//null-@{bin}/dbus-update-activation-environment[0m [33mfile_mmap[0m [35m@{bin}/dbus-update-activation-environment[0m comm=dbus-update-act requested_mask=[1;31mr[0m denied_mask=[1;31mr[0m
[1;32mALLOWED[0m [34mgeoclue[0m [33mfile_inherit[0m[35m owner[0m [35m@{user_share_dirs}/sddm/xorg-session.log[0m comm=agent requested_mask=[1;31mw[0m denied_mask=[1;31mw[0m
[1;32mALLOWED[0m [34mdbus-accessibility[0m [33mopen[0m[35m owner[0m [35m/tmp/xauth_jcSShq[0m comm=at-spi-bus-laun requested_mask=[1;31mr[0m denied_mask=[1;31mr[0m
[1;32mALLOWED[0m [34mxfs_spaceman[0m [33mcapable[0m comm=xfs_spaceman capability=2 capname=dac_read_search

roddhjav commented on apparmor.d/groups/lxqt/lxqt-about:
```diff
@@ -0,0 +1,31 @@
+
+  /usr/share/icons/* 			r,
+  /usr/share/icons/{,**}  		r,
+  /usr/share/desktop-directories/* 	r,
+  /usr/share/desktop-directories/{,**}  r,
```
The second rule include the first rule.

roddhjav commented on apparmor.d/groups/lxqt/lxqt-about:
```diff
@@ -0,0 +1,31 @@
+      
+  @{exec_path}			mr,
+
+  /usr/share/icons/* 			r,
+  /usr/share/icons/{,**}  		r,
```
`/usr/share/icons/{,**}` is already included in `abstractions/freedesktop.org` (wich is in `abstractions/lxq`).

roddhjav commented on apparmor.d/groups/browsers/falkon:
```diff
@@ -0,0 +1,215 @@
+  include <abstractions/user-download-strict>
+  include <abstractions/user-read-strict>
+
+  network inet dgram,		# essential
+  network inet stream,		# essential
+
+  network inet6 stream,   	# Not needed
+  network inet6 dgram,    	# Not needed
+  network inet raw,      	# Not needed
+  network inet6 raw,		# Not needed
+  network netlink raw,		# Not needed
+  network packet dgram,		# Not needed
```
If there are not needed, then do not put them...

The following int6 are needed for IPV6: 
```
  network inet6 dgram,
  network inet6 stream,
```

Besanon commented:
Hi,

you put the activated processes here, runsv will not claim any more
simple, but you have to track this. I will put a comment in this profile to explain this

Am 07.06.2024 um 13:52, Alex ***@***.***> schrieb:

@roddhjav commented on this pull request.
In apparmor.d/groups/runit/runsv:> +  @{bin}/sv 		 rPx,+  @{bin}/vlogger         rPx,+  @{bin}/udevadm 	 rCx -> udevadm, +  @{bin}/tlp             rPx,+  @{bin}/readlink        rix,+  @{bin}/ethtool         rix,+  @{bin}/agetty          rPx,+  @{bin}/id              rPx,+  @{bin}/rsyslogd        rPx,+  @{bin}/iw              rPx,+  @{bin}/cupsd           rPx,+  @{bin}/dhcpcd          rPx,+  @{bin}/udevd           rPx,+  @{bin}/dbus-daemon     rPx,+  @{bin}/auditd          rPx,+  @{bin}/chronyd         rPx,+  @{bin}/NetworkManager  rPx,+  @{bin}/mount  	 rPx,+  @{bin}/sddm            rPx,+  @{bin}/pause           rix,+  @{bin}/install         rix,+  @{bin}/chpst           rPx,+  @{bin}/mkdir           rix,+  @{bin}/mktemp          rix,+  @{bin}/dbus-send       rix, # alt: rix+  @{bin}/utmpset	 rix, # alt: rix+  @{lib}exec/elogind/elogind          rPx,+  @{lib}exec/elogind/elogind.wrapper  rPx,  # alt: rix+  @{bin}/bash		 rPx, # alt: rix+  @{bin}/tr		 rPx, # alt:rix+  @{bin}/rm		 rix,+  @{bin}/touch		 rix,+  @{bin}/flock		 rix,+  @{bin}/cat		 rix,+  @{bin}/grep		 rPx, # alt:rix+  @{bin}/mountpoint 	 rix,+  @{bin}/systemctl       rCx -> systemctl,
I don't know runit. Therefore I can't says which process is doing what. But looking at this list, to me it looks like the list of service started by your system.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

-- 
Sent with https://mailfence.com  
Secure and private email

Besanon commented:
ok, replaced

Am 07.06.2024 um 13:42, Alex ***@***.***> schrieb:

@roddhjav commented on this pull request.
In apparmor.d/groups/browsers/falkon:> +  network inet dgram,		# essential+  network inet stream,		# essential++  network inet6 stream,   	# Not needed+  network inet6 dgram,    	# Not needed+  network inet raw,      	# Not needed+  network inet6 raw,		# Not needed+  network netlink raw,		# Not needed+  network packet dgram,		# Not needed
If there are not needed, then do not put them...
The following int6 are needed for IPV6:  network inet6 dgram,  network inet6 stream,
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

-- 
Sent with https://mailfence.com  
Secure and private email

roddhjav commented:
> dbus                 wants to exec dbus-update-activation-environment

What `dbus` profile? there is no such thing as a `dbus` profile (see https://apparmor.pujol.io/development/dbus/#profiles).
Usually this is more found as a subprofile, and it is quite common. See for example: https://github.com/roddhjav/apparmor.d/blob/b66274b2ca4a36873cdf9f7ae5f7ef2133e94b9d/apparmor.d/groups/display-manager/x11-xsession#L109

> sddm-greeter    wants to receive signal from sddm

This seems normal

> the falkon profile is adapted.

Some rules seems taken from firefox/chromium whereas falkon does not seems to be based on either of them.

> As far as i can see, there are issues with dbus-update-activation in startlxqt and lxqt-notificationd. 

According to your logs. startlxqt and lxqt-notificationd need the `attach_disconnected` flag.

> Regarding runit profiles I have some doubts. As the init system is closely related to the rest of the system, adding support for another one is a lot of work and I simply don't have the resources for it.

Regarding runit profiles (same for elogind), as I have said, I am not going to merge them.

Also, I would highly recommend you to edit the profile on your system and to push your changes and not directly edit the files on Github. (see "How to contribute" on https://apparmor.pujol.io/development/). It will allow you to add, remove, edit without having to create a commit for every file changed.

nobody43 commented on apparmor.d/groups/browsers/falkon:
```diff
@@ -0,0 +1,215 @@
+  owner /tmp/@{name}/* rwk,
+  owner /tmp/@{rand6}.tmp r,
+  owner /tmp/falkon-*/ rw,
+  owner /tmp/falkon-*/* rwk,
```
Could it be covered by `/tmp/falkon-??????/`?

nobody43 commented on apparmor.d/groups/lxqt/lxqt-config-notificationd:
```diff
@@ -0,0 +1,34 @@
+  owner @{user_config_dirs}/lxqt/ 	r,
+  owner @{user_config_dirs}/lxqt/** rwkl -> @{user_config_dirs}/lxqt/#@{int},
+
+  owner /tmp/{,**}  r,
```
Are these paths really unpredictable?

nobody43 commented on apparmor.d/groups/lxqt/lxqt-panel:
```diff
@@ -0,0 +1,82 @@
+  include <abstractions/nameservice-strict>
+
+  network inet dgram,
+  network inet stream,          `
```
Syntax error.

nobody43 commented on apparmor.d/groups/lxqt/lxqt-session:
```diff
@@ -0,0 +1,102 @@
+  owner @{HOME}/.config/			r,
+  owner @{HOME}/.config/autostart/		r,
+  owner @{HOME}/.config/autostart/*		rw,
+  owner @{HOME}/.config/mimeapps.list*		rw,
```
`@{user_config_dirs}/mimeapps.list{,.@{rand6}}`?

nobody43 commented on apparmor.d/groups/lxqt/lxqt-session:
```diff
@@ -0,0 +1,102 @@
+
+  /dev/tty                              rw,
+  /dev/tty[0-9]*                        rw,
+  /dev/pts/[0-9]*                       rw,
```
`abstractions/consoles`

nobody43 commented on apparmor.d/profiles-a-f/elogind:
```diff
@@ -0,0 +1,191 @@
+  /usr/share/p11-kit/modules/* r,
+
+  # gnome-keyring pkcs11 module
+  owner @{run}/user/[0-9]*/keyring*/pkcs11 rw,
```
`@{run}/user/@{uid}/keyring{,s}/pkcs11`?

nobody43 commented on apparmor.d/profiles-a-f/elogind:
```diff
@@ -0,0 +1,191 @@
+  owner /var/log/wtmp rwk,
+  owner /dev/shm/ r,
+  owner /dev/dri/card@{int} rw,
+  owner /dev/tty@{int} rw,
```
Redundant.

nobody43 commented:
Like already said: abstractions and tunables aren't leveraged, `/tmp/{,**}` access is too broad.
And please edit files of such quantity - in batch locally with git commits. Web edits produce to much notification spam.

Besanon commented:
Good Morning,

i have spent some hours to remove the /tmp{,**}, and yes i tried to get into working with github cli but i didn`t succeed so far

xbps-install -S github-cli

mkdir projekte/Besanon
cd projekte/Besanon
gh repo clone Besanon/apparmor.d
cd apparmor.d
gh repo set-default Besanon/apparmor.d
git checkout -b BesanonLocal

... do some editing in console ...

git commit -a -m "Test"
gh auth login --with-token < /opt/mytoken.txt  # mytoken.txt created in github profile
git commit -a -m "Test"
git push origin Besanonlocal  -> requests user & pw but didn`t accept my credentials (=token???]

Should this be the way to commit changes? Its a very embarassing description in github doku. 

Am 09.06.2024 um 20:31, nobody43 ***@***.***> schrieb:

Like already said: abstractions and tunables aren't leveraged, /tmp/{,**} access is too broad.
And please edit files of such quantity - in batch locally with git commits. Web edits produce to much notification spam.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

-- 
Sent with https://mailfence.com  
Secure and private email
